src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -118,6 +118,46 @@ SpatialPooler::SpatialPooler()
   version_ = 2;
 }
 
+SpatialPooler::SpatialPooler(vector<UInt> inputDimensions,
+                                  vector<UInt> columnDimensions,
+                                  UInt potentialRadius,
+                                  Real potentialPct,
+                                  bool globalInhibition,
+                                  Real localAreaDensity,
+                                  UInt numActiveColumnsPerInhArea,
+                                  UInt stimulusThreshold,
+                                  Real synPermInactiveDec,
+                                  Real synPermActiveInc,
+                                  Real synPermConnected,
+                                  Real minPctOverlapDutyCycles,
+                                  Real minPctActiveDutyCycles,
+                                  UInt dutyCyclePeriod,
+                                  Real maxBoost,
+                                  Int seed,
+                                  UInt spVerbosity,
+                                  bool wrapAround) : SpatialPooler::SpatialPooler() 
+{
+  initialize(			  inputDimensions,
```

Comment:
This still doesn't set the version. Can you add the following right before the call to initialize:

```
SpatialPooler();
```

That way this function exactly matches calling the constructor with no args followed by `initialize`.